blocks/monitoring/evaluators.py

```diff
@@ -262,9 +262,15 @@ def _compile(self):
         if self.theano_buffer.accumulation_updates:
             updates = OrderedDict()
             updates.update(self.theano_buffer.accumulation_updates)
-            if self.updates:
-                updates.update(self.updates)
             inputs += self.theano_buffer.inputs
+        if self.updates:
```

Comment:
This change should not have been merged without tests, since it would be pretty easy to write a test for it. Also it seems to me that always initializing `updates` in the line 261 to `OrderedDict()` would do the job. Am I right?